File: GO/0006-zigzag-conversion.go
func convert(s string, numRows int) string {
    ans := make([][]rune, numRows)
    i:= 0
    for i < len(s) {
        j := 0
        for j < numRows && i < len(s) {
            ans[j] = append(ans[j], rune(s[i]))
            j++
            i++
        }

        j = numRows - 2
        for j > 0 && i < len(s) {
            ans[j] = append(ans[j], rune(s[i]))
            j--
            i++
        }
    }
    result := ""
    for _, row := range ans {
        result += string(row)
    }
    return result
}

File: Python/0006-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        ans = [[] for _ in range(numRows)]
        i = 0
        while i < len(s):
            j = 0
            while j < numRows and i < len(s):
                ans[j].append(s[i])
                j += 1
                i += 1
                    
            j = numRows - 2
            while j > 0 and i < len(s):
                ans[j].append(s[i])
                j -= 1
                i += 1

        result = ""
        for row in ans:
            result += "".join(row)

        return result
